src/main/java/io/quarkus/bot/CancelWorkflowOnClosedPullRequest.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.bot;
+
+import io.quarkiverse.githubapp.event.PullRequest;
+import io.quarkus.bot.config.QuarkusBotConfig;
+import io.quarkus.bot.workflow.WorkflowConstants;
+import org.jboss.logging.Logger;
+import org.kohsuke.github.GHEventPayload;
+import org.kohsuke.github.GHPullRequest;
+import org.kohsuke.github.GHWorkflowRun;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+class CancelWorkflowOnClosedPullRequest {
+    private static final Logger LOG = Logger.getLogger(CancelWorkflowOnClosedPullRequest.class);
+
+    @Inject
+    QuarkusBotConfig quarkusBotConfig;
+
+    public void onClose(@PullRequest.Closed GHEventPayload.PullRequest pullRequestPayload) throws IOException {
+
+        GHPullRequest pullRequest = pullRequestPayload.getPullRequest();
+
+        List<GHWorkflowRun> ghWorkflowRuns = pullRequest.getRepository()
+                .queryWorkflowRuns()
+                .branch(pullRequest.getHead().getRef())
+                .list()
+                .toList()
+                .stream()
+                .filter(workflowRun -> WorkflowConstants.QUARKUS_CI_WORKFLOW_NAME.equals(workflowRun.getName()) &&
+                        workflowRun.getHeadRepository().getOwnerName()
+                                .equals(pullRequest.getHead().getRepository().getOwnerName()))
+                .filter(workflowRun -> workflowRun.getStatus() == GHWorkflowRun.Status.QUEUED
+                        || workflowRun.getStatus() == GHWorkflowRun.Status.IN_PROGRESS)
+                .collect(Collectors.toList());
+
+        for (GHWorkflowRun workflowRun : ghWorkflowRuns) {
+            if (!quarkusBotConfig.isDryRun()) {
+                workflowRun.cancel();
+            } else {
+                LOG.info("Workflow run #" + workflowRun.getId() + " - Cancelling as pull request #" + pullRequest.getId()
```

Comment:
```suggestion
                LOG.info("Workflow run #" + workflowRun.getId() + " - Cancelling as pull request #" + pullRequest.getNumber()
```
(id is the internal id)